spp_base_gis/models/base.py

```diff
@@ -93,3 +133,192 @@ def get_edit_info_for_gis_column(self, column):
+        function takes this list of features and wraps them in a GeoJSON
+        :type features: list
+        :return: A dictionary is being returned with the keys "type" and "features". The value of the
+        "type" key is set to "FeatureCollection", and the value of the "features" key is set to the
+        input parameter `features`, which is a list of features.
+        """
+        return {"type": "FeatureCollection", "features": features if isinstance(features, list) else [features]}
+
+    @api.model
+    def get_field_type_from_layer_type(self, layer_type):
+        """
+        The function `get_field_type_from_layer_type` maps a layer type to a corresponding field type
+        for geographic data, raising a UserError if the layer type is invalid.
+
+        :param layer_type: The `get_field_type_from_layer_type` function takes a `layer_type` as input
+        and returns the corresponding field type based on a mapping defined in the `layer_type_mapping`
+        dictionary. The mapping assigns a field type to each layer type - "point", "line", and "polygon"
+        :return: The function `get_field_type_from_layer_type` returns the corresponding field type
+        based on the given `layer_type`. If the `layer_type` is "point", it returns "geo_point". If the
+        `layer_type` is "line", it returns "geo_line". If the `layer_type` is "polygon", it returns
+        "geo_polygon". If the `layer_type` is not one
+        """
+        layer_type_mapping = {
+            "point": "geo_point",
+            "line": "geo_line",
+            "polygon": "geo_polygon",
+        }
+
+        try:
+            return layer_type_mapping[layer_type]
+        except KeyError as e:
+            raise UserError(_("Invalid layer type %s") % layer_type) from e
+
+    @api.model
+    def raw_postgis_sql_query(self, postgis_query):
+        """
+        The function `raw_postgis_sql_query` executes a raw PostGIS SQL query on a specified table and
+        returns the results.
+
+        :param postgis_query: The `postgis_query` parameter in the `raw_postgis_sql_query` method is
+        expected to be a PostGIS query that will be used in the `WHERE` clause of the SQL query
+        constructed within the method. This query should be a valid PostGIS query that filters the
+        records in the database
+        :return: The method `raw_postgis_sql_query` is returning the result of the SQL query executed on
+        the database using the provided `postgis_query`. The method executes the query using
+        `self.env.cr.execute(query)` and then fetches all the results using `self.env.cr.fetchall()`.
+        The method will return a list of tuples where each tuple represents a row of the result set.
+        """
+        query = f"SELECT id FROM {self._table} WHERE {postgis_query}"
+        self.env.cr.execute(query)
+        return self.env.cr.fetchall()
```

Comment:
This seems risky with SQL Injection. Can we allow this with special right? By default, nobody should have it.

Reply:
Could we do a "safer" version that take a domain as parameter?

Reply:
noted on this. i'll work on this today

Reply:
Done.